### ISSUE TITLE
"search" ability scenario

### ISSUE THREAD

**Issue by h0111in** [h0111in/PokemonGoBack#51]
@z-mehta, please check the program and write the correct scenario's steps for me, if it is wrong.

**Comment by z-mehta** [comment]
actually, the search ability should work like this -
1. it search your deck for energy cards(say : 4 energy cards)
2. then reveal them to opponent(optional in our case)
3,add them to your hand section
4. shuffle the deck afterwards

**Comment by z-mehta** [comment]
search in special case may use a condition....or sometimes search for varying no. of energy cards

**Comment by z-mehta** [comment]
actually search ability basically used for searching in the deck(your deck or opponent)........you can either see opponent's deck cards........obtain cards from our deck. after searching the deck will be shuffled

**Comment by z-mehta** [comment]
@h0111in can you specify which exactly ability you are talking about related to search scenario?

**Comment by h0111in** [comment]
EX: abilities.txt :for the following line, what exactly it means?
search:target:choice:your-pokemon:cat:basic:source:deck:filter:evolves-from:target:last:1,shuffle:target:you

**Comment by z-mehta** [comment]
#45 related issue

**Comment by z-mehta** [comment]
this ability follows the below steps:

1. search your deck for the pokemon cards which evolves from your active card.
2. check for condition: if there are more than 2 cards which can evolve from your active pokemon card, choose only one among them.
3.evolve happens.
4.shuffle your deck.

Note: we can use trainer card once per turn, like after using trainer card the turn should pass. Also, when using this trainer card, it should placed next to your active pokemon card until the ability happen. after ability happens, move this card to discard section.
in other words, when we click on this card's ability, then it should directly placed next to active card and when ability is done shift it to discard area.

**Comment by z-mehta** [comment]
5. pass the turn to opponent.
@h0111in  look into it

**Comment by h0111in** [comment]
search has bean implemented again. please check it.
